miqa/learning/nn_training.py

```diff
@@ -318,8 +318,8 @@ def apply_transform(self, subject: torchio.Subject) -> torchio.Subject:
             transformed_subject = super().apply_transform(subject)
 
             # make sure we don't have negative intensities after adding noise
-            transformed_subject.img.data = torch.clamp(
-                transformed_subject.img.data, min=0.0, max=1.0
+            transformed_subject.img.set_data(
+                torch.clamp(transformed_subject.img.data, min=0.0, max=1.0)
```

Comment:
Since you're getting rid of deprecation warnings in this PR, is it possible to fix the warnings we get in CI tests that say that "the `scipy.ndimage.morphology` namespace is deprecated"?

Reply:
I created #344 to track this.